register_forte_options.py

```diff
@@ -117,6 +118,15 @@ def register_mrcino_options(forte_options):
         "actice_docc and restricted_docc"
         "or not")
 
+def register_embedding_options(forte_options):
+        forte_options.add_bool("EMBEDDING", False, "Whether to perform embedding partition and projection")
+        forte_options.add_str("CUTOFF_BY", "THRESHOLD", "Cut off by: threshold or number.")
+        forte_options.add_double("THRESHOLD", 0.5, "Projector eigenvalue threshold, 0.5 as default")
+        forte_options.add_str("REFERENCE", "HF", "HF, ROHF, UHF(not implemented), MCSCF, CASSCF")
```

Comment:
Here there are several potential improvements:

1. Use the form of add_str with with the list of allowed values, to avoid having to check user input. This way you also do not need to say `UHF(not implemented)`
2. Do we want `HF` to be the default here? It would be better to make "CASSCF" default and revert to "HF" if there are no active orbitals. Otherwise someone inexperienced using the code might end up running the wrong computation.
3. Is there a difference between MCSCF and CASSCF?